### PR TITLE
[Feature][RestAPI] Add overview api

### PR DESCRIPTION
**Pull request by liugddx**
### Purpose of this pull request

{
  "projectVersion": "2.3.5",
  "gitCommitAbbrev": "DeadD0d0",
  "totalSlot": "0",
  "unassignedSlot": "0",
  "runningJobs": "0",
  "finishedJobs": "0",
  "failedJobs": "0",
  "cancelledJobs": "0",
  "works": "1"
}

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/AbstractResourceManager.java [hunk 1]
```diff
@@ -223,4 +223,16 @@ public List<SlotProfile> getUnassignedSlots() {
...
+
+    @Override
+    public List<SlotProfile> getAssignedSlots() {
+        return registerWorker.values().stream()
+                .flatMap(workerProfile -> Arrays.stream(workerProfile.getAssignedSlots()))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public int workCount() {
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/AbstractResourceManager.java]:
```suggestion
    public int workerCount() {
```

File: seatunnel-examples/seatunnel-engine-examples/src/main/java/org/apache/seatunnel/example/engine/SeaTunnelEngineExample.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ public static void main(String[] args)
         clientCommandArgs.setJobName(Paths.get(configFile).getFileName().toString());
         // Change Execution Mode to CLUSTER to use client mode, before do this, you should start
         // SeaTunnelEngineServerExample
-        clientCommandArgs.setMasterType(MasterType.LOCAL);
+        clientCommandArgs.setMasterType(MasterType.CLUSTER);
```
Comment (Hisoka-X) [inline seatunnel-examples/seatunnel-engine-examples/src/main/java/org/apache/seatunnel/example/engine/SeaTunnelEngineExample.java]:
```suggestion
        clientCommandArgs.setMasterType(MasterType.LOCAL);
```

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/opeartion/GetOverviewOperation.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                finishedJob.values().stream()
+                        .filter(
+                                jobState ->
+                                        jobState.getJobStatus()
+                                                .name()
+                                                .equals(JobStatus.CANCELED.toString()))
+                        .count());
+        overviewInfo.setWorks(resourceManager.workCount());
+        overviewInfo.setFinishedJobs(
+                finishedJob.values().stream()
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/opeartion/GetOverviewOperation.java]: I find this part show up twice. Please do some refactor.

**Review by Hisoka-X — COMMENTED** [review]
Thanks @liugddx ! I like this feature.

**Review by Hisoka-X — DISMISSED** [review]

**Comment by hailin0** [comment]
Please fix ci error
image

**Comment by Hisoka-X** [comment]
image

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**